comp/otelcol/otlp/components/exporter/serializerexporter/exporter.go

```diff
@@ -127,15 +133,16 @@ func translatorFromConfig(
 
 // NewExporter creates a new exporter that translates OTLP metrics into the Datadog format and sends
 func NewExporter(
-	set component.TelemetrySettings,
+	params exporter.Settings,
 	attributesTranslator *attributes.Translator,
 	s serializer.MetricSerializer,
 	cfg *ExporterConfig,
 	enricher tagenricher,
 	hostGetter SourceProviderFunc,
 	statsIn chan []byte,
+	createConsumer createConsumerFunc,
 ) (*Exporter, error) {
-	tr, err := translatorFromConfig(set, attributesTranslator, cfg.Metrics.Metrics, hostGetter, statsIn)
+	tr, err := translatorFromConfig(params.TelemetrySettings, attributesTranslator, cfg.Metrics.Metrics, hostGetter, statsIn)
```

Comment:
You will need different translator for Agent (WithOTelPrefix) vs. collector (WithRemapping):
http://github.com/DataDog/datadog-agent/blob/4577817fb865565f8dd5243f243543ab3f866f56/comp/otelcol/otlp/components/exporter/serializerexporter/exporter.go#L94

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1c3743853f0ec2de78bc7873ed04d900ebe8b1a7/exporter/datadogexporter/metrics_exporter.go#L74